nautobot/dcim/migrations/0030_migrate_region_and_site_data_to_locations.py

```diff
@@ -191,6 +194,48 @@ def migrate_site_and_region_data_to_locations(apps, schema_editor):
             cc.locations.add(*region_locs)
             cc.save()
 
+        # Custom Field
+        CustomField = apps.get_model("extras", "customfield")
+        custom_fields = CustomField.objects.filter(content_types__in=[region_ct])
+        for cf in custom_fields:
+            cf.content_types.add(ContentType.objects.get_for_model(Location))
+            cf.save()
+
+        region_locs = Location.objects.filter(location_type=region_lt).exclude(name="Global Region")
+        for location in region_locs:
+            region = Region.objects.get(name=location.name)
+            location._custom_field_data = region._custom_field_data
+        Location.objects.bulk_update(region_locs, ["_custom_field_data"], 1000)
+
+        # Relationship
+        Relationship = apps.get_model("extras", "relationship")
+        RelationshipAssociation = apps.get_model("extras", "relationshipassociation")
+        src_relationships = Relationship.objects.filter(source_type=region_ct)
+        for relationship in src_relationships:
+            relationship.source_type = ContentType.objects.get_for_model(Location)
+            relationship.save()
+            relationship_associations = RelationshipAssociation.objects.filter(relationship=relationship)
+            relationship_associations.update(source_type=ContentType.objects.get_for_model(Location))
+            for relationship_association in relationship_associations:
```

Comment:
I wonder, would it be possible and/or more efficient to do a single loop over all of `RelationshipAssociation.objects.filter(relationship__source_type=region_ct)` instead of handling them per-Relationship?